### DISCUSSION TITLE
Client for C#

### DISCUSSION THREAD

**Discussion by brendofreitas** [mosparo/mosparo#111] [Ideas]
I'm having some difficulties integrating mosparo with C#, an example of a client with C# would be a great idea,
I'm having some header invalid issues.

If you have some C# cases I would like to take a look

If not, an example would be a great idea, it would gain a lot of users.

**Comment by zepich** [comment]
Hi @brendofreitas 

Thank you very much for your post.

I will prepare an example. Can you tell me what you use as a framework, HTTP client, and so on so I can try to test it in a similar environment?

Thank you very much!

**Reply by brendofreitas** [reply]
I currently use the .NET Framework 4.7.2 web, I believe that with this framework everyone can take a step 0 and use it as a basis for other versions and thus have a lot of use, C # has a lot of users, thus greatly expanding the usefulness of the tool.

Thank you very much for answering promptly, if you can I will be very grateful and looking forward to using it!

**Reply by zepich** [reply]
Hi @brendofreitas 

I've set up a project with C# and .NET Framework and added a test form and the required logic to verify the form submission with mosparo.

The main code is visible here: https://github.com/zepich/exmple-mosparo-csharp/blob/master/mosparo-example/Testform.aspx.cs

Does that example reflect your setup, or did I use the wrong one?

Does it help you with your problem?

Please remember that this is just a simple example of the API request, not the complete verification process. After the verification is done, the verification signature should be verified (which I skipped in my example), and it must be made sure that the bypass protection is set up correctly (see here https://documentation.mosparo.io/docs/integration/bypass_protection).

I wish you a nice day and look forward to hearing from you.

**Reply by brendofreitas** [reply]
It helped a lot, thanks.

Asp Classico is not widely used but the way it was made made clear some points that didn't need to be done.

1- No need to reorganize the form data.

2- You didn't need to put everything in lower case.

3- In all fields, you must replace CRLF line breaks with LF line breaks (convert \r\nto \n).

I made a client and put it as Action Filters called [MosparoCaptchaValidator]

This makes it easier to use the implementation, I will go up in the future if you want to use or base it on.

**Reply by zepich** [reply]
@brendofreitas Perfect. Thank you very much for your feedback.

I'm happy that it's working for you. The action filter sounds like a good idea.

Do you plan to release the client to the public?
